x/auctions/keeper/utils.go

```diff
@@ -0,0 +1,22 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/evmos/v19/utils"
+)
+
+// removeBaseCoinFromCoins returns an sdk.Coins removing the
+// base denom from the input.
+func removeBaseCoinFromCoins(coins sdk.Coins) sdk.Coins {
+	remainingCoins := sdk.NewCoins()
+	// TODO: check if evmos tokens are accumulated when the module has more than the bid.
+	for _, coin := range coins {
+		if coin.Denom != utils.BaseDenom {
+			remainingCoins = remainingCoins.Add(coin)
+		}
+	}
+	return remainingCoins
```

Comment:
actually I wonder if it's quicker to do this instead? not sure but maybe because there is only one `Sub` instead of many different `Add`


```suggestion
func removeBaseCoinFromCoins(coins sdk.Coins) sdk.Coins {
	// TODO: check if evmos tokens are accumulated when the module has more than the bid.
	
    return coins.Sub(sdk.Coin{
        Denom: utils.BaseDenom,
        Amount: coins.AmountOfNoDenomValidation(utils.BaseDenom),
    })
```

Reply:
Not sure about this since you have hidden iterations also inside the `Sub` method and the `AmountOfNoDenom...` is using a `Find` inside. We should benchmark it but I'm pretty sure that given the small size of coins it will be not very relevant. I will keep the code we have at the moment since it is more intuite imo.

Reply:
>the code we have at the moment since it is more intuite imo

The whole reason why I commented on this was that I found it very unintuitive to not just use the available methods from the Cosmos SDK haha.

I also tested with a quick benchmark:

```go
func BenchmarkRemoveBaseCoinFromCoins(b *testing.B) {
	coins := sdk.NewCoins(
		sdk.NewCoin("foo", math.NewInt(100)),
		sdk.NewCoin("bar", math.NewInt(200)),
		sdk.NewCoin("baz", math.NewInt(300)),
	)

	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		auctionskeeper.RemoveBaseCoinFromCoins(coins)
	}
}
```

and my suggestion won :) 

current (manual): `BenchmarkRemoveBaseCoinFromCoins-12      5384088              1116 ns/op`
suggestion (sub): `BenchmarkRemoveBaseCoinFromCoins-12      8682234               684.7 ns/op`